cli/flox/src/utils/metrics.rs

```diff
@@ -217,75 +211,107 @@ async fn push_metrics(mut metrics: MetricsBuffer, uuid: Uuid) -> Result<()> {
         })
         .collect::<Result<Vec<serde_json::Value>>>()?;
 
-    reqwest::Client::new()
+    debug!("Sending metrics to {}", &*METRICS_EVENTS_URL);
+    debug!("Metrics: {events:?}", events = events);
+
+    let req = reqwest::Client::new()
         .put(&*METRICS_EVENTS_URL)
         .header("content-type", "application/json")
         .header("x-api-key", METRICS_EVENTS_API_KEY)
         .header("user-agent", format!("flox-cli/{}", version))
         .json(&events)
-        .send()
-        .await?;
+        .send();
 
-    metrics.clear().await;
+    let handle = tokio::runtime::Handle::current();
+    let _guard = handle.enter();
+    futures::executor::block_on(req).context("could not send to telemetry backend")?;
 
+    metrics.clear()?;
     Ok(())
 }
 
-#[derive(Default, Debug)]
-struct MetricsBuffer(Option<File>, VecDeque<MetricEntry>);
+/// A representation of the metrics buffer
+///
+/// The metrics buffer is a file that contains a list of metrics entries.
+/// It is used to store metrics for a period of time
+/// and then push them to the server.
+///
+/// An instance of this struct represents the metrics buffer file and its contents.
+/// While the metrics buffer is being used, it is locked to avoid data corruption.
+/// Thus, a [MetricsBuffer] instance should be short-lived
+/// to avoid blocking other processes.
+#[derive(Debug)]
+struct MetricsBuffer {
+    /// The file where the metrics buffer is stored
+    storage: File,
+    /// The lock file for the metrics buffer.
+    /// Used to avoid concurrent writes to the metrics buffer file.
+    _file_lock: LockFile,
+    buffer: VecDeque<MetricEntry>,
+}
 impl MetricsBuffer {
-    /// Reads the metrics buffer from the given file
-    async fn read_from_file(mut file: File) -> Result<Self> {
-        let mut buffer_json = String::new();
-        file.read_to_string(&mut buffer_json).await?;
-
-        let buffer_iter = serde_json::Deserializer::from_str(&buffer_json)
-            .into_iter::<MetricEntry>()
-            .filter_map(|x| x.ok())
-            .collect();
-
-        Ok(MetricsBuffer(Some(file), buffer_iter))
-    }
-
     /// Reads the metrics buffer from the cache directory
-    async fn read(config: &Config) -> Result<Self> {
-        // dont create a metrics buffer if metrics are disabled anyway
-        if config.flox.disable_metrics {
-            return Ok(Default::default());
-        }
-
-        let cache_dir = &config.flox.cache_dir;
-
+    fn read(cache_dir: &Path) -> Result<Self> {
         let mut metrics_lock = LockFile::open(&cache_dir.join(METRICS_LOCK_FILE_NAME))?;
-        tokio::task::spawn_blocking(move || metrics_lock.lock()).await??;
+        metrics_lock.lock()?;
```

Comment:
suggestion nonblocking: add a comment about when `unlock` happens. I assume the file handle is dropped, but it's non-obvious since it's moved out of this function. And you said holding the lock should be short lived